extension/icu/include/icu-datefunc.hpp

```diff
@@ -58,6 +58,14 @@ struct ICUDateFunc {
 	static int32_t ExtractField(icu::Calendar *calendar, UCalendarDateFields field);
 	//! Subtracts the field of the given date from the calendar
 	static int64_t SubtractField(icu::Calendar *calendar, UCalendarDateFields field, timestamp_t end_date);
+	//! Adds the timestamp and the interval using the calendar
+	static timestamp_t Add(icu::Calendar *calendar, timestamp_t timestamp, interval_t interval);
+	//! Subtracts the interval from the timestamp using the calendar
+	static timestamp_t Sub(icu::Calendar *calendar, timestamp_t timestamp, interval_t interval);
+	//! Subtracts the latter timestamp from the former timestamp using the calendar
+	static interval_t Sub(icu::Calendar *calendar, timestamp_t end_date, timestamp_t start_date);
+	//! Add UTC offset to the naive timestamp
```

Comment:
> Unfortunately, that is not exactly what this does. What it does is pull out the bin values from the timestamp _assuming it is an instant_, uses those values to construct an ICU timestamp and then converts that back to a DuckDB instant. The addition part doesn't really work around DST because the bin values are ambiguous.

Pasted this to comment :P

Reply:
Thanks - please resolve!